contracts/DegenLockToken.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+/**
+ * @notice Lock Degen for a set amount of time.
+ * @custom:security-contact jacek@degen.tips
+ */
+contract DegenLockToken is ERC20, Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    mapping(address account => uint256) private _depositTimestamps;
+
+    /**
+     * @dev Name of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_NAME = "Locked Degen";
+
+    /**
+     * @dev Symbol of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_SYMBOL = "LDEGEN";
+
+    /**
+     * @dev The ERC20 token to be locked
+     */
+    address public constant TOKEN = 0xfee293840D23B0B2De8C55e1Cf7A9F01C157767c;
```

Comment:
The type can be `IERC20` instead of `address` since `TOKEN` seems to always be casted to `IERC20` before being used below.

Reply:
Fixed.